### PR TITLE
Dismiss Completion when Signature Help is explicitly invoked

### LINKED ISSUE #42484: Explicit invocation of Signature Help should cancel any existing Completion List session
In https://github.com/dotnet/roslyn/pull/42035, we introduce the completion list in argument lists in C#. This can cause conflict between Signature Help & Completion, so it was requested that any explicit invocation of Signature Help cancel existing Completion sessions, so that SigHelp gets focus and the arrow keys move through overloads as expected.

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes #42484

### CODE REVIEW

**Comment by dpoeschl** [comment]
@CyrusNajmabadi Implemented this on our side as discussed. Let me know if you have concerns with this approach.

File: src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller_InvokeSignatureHelp.cs [hunk 1]
```diff
@@ -28,6 +28,11 @@ void IChainedCommandHandler<InvokeSignatureHelpCommandArgs>.ExecuteCommand(Invok
                 return;
             }
 
+            // Dismiss any completion sessions when Signature Help is explicitly invoked. The
+            // primary motivation for this behavior is when both Signature Help and Completion
+            // are activated at the same time, for example in argument lists.
+            _completionBroker.GetSession(args.TextView)?.Dismiss();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller_InvokeSignatureHelp.cs]: nice!  glad this was so simple.  perhaps expand on the message to make it clear that there are cases where both come up *implciitly*, but this is the users was to *explicitly* state waht is important to them.

Comment (dpoeschl) [reply src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller_InvokeSignatureHelp.cs]: Updated the comment.

**Comment by dpoeschl** [comment]
@CyrusNajmabadi Thanks for taking a look. I've taken this out of Draft state and added an integration test. Can you take another look and sign off if appropriate? Thanks!

File: src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller_InvokeSignatureHelp.cs [hunk 2]
```diff
@@ -28,6 +28,13 @@ void IChainedCommandHandler<InvokeSignatureHelpCommandArgs>.ExecuteCommand(Invok
                 return;
             }
 
+            // Dismiss any Completion sessions when Signature Help is explicitly invoked.
+            // There are cases when both show up implicitly, for example in argument lists
+            // when the user types the `(`. If both are showing and the user explicitly
+            // invokes Signature Help, they are requesting that the Signature Help control 
+            // be the focused one. Closing an existing Completion session accomplishes this.
+            _completionBroker.GetSession(args.TextView)?.Dismiss();
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller_InvokeSignatureHelp.cs]: FYI @AmadeusW that we're using the editor's completion broker for this.

**Comment by dpoeschl** [comment]
@jinujoseph `master` is Preview 3 now, right? Is it too late to push for this to get into Preview 2?

**Pull request merged**